Add an express-session compatible store adapter

atuttle opened the pull request:
Also adds some documentation about the different approach that this adapter takes to the readme. This PR does not break backwards compatibility, simply adds a different approach to storing the data which is compatible with express-session.

I intend to add core docs (in the empty section I added for them) for the standard Sidecar approach soon. Not sure yet if that means before or after this PR is merged. :)

atuttle committed:
add express-session compatible store adapter
Also adds some documentation about the different approach that this adapter takes to the readme
I intend to add core docs for the standard Sidecar approach soon

ryanguill commented on com/express_redis_session_store.cfc:
```diff
@@ -0,0 +1,225 @@
+			if (len(err)) throw(err);
+			//result = redis.expire(prefix & sessionID, expires);
+			set(sessionID, "_session_expires", expires);
+			redis.zadd(prefix & "_session_expires", expires, sessionID);
```
@atuttle I worry that you shouldnt use the `_session_expires` stuff like we do in the default redis store.   This will work fine as long as sidecar is the only one that is expiring sessions, but if express-session does then it wont be touching or expiring things properly.  I think you need to remove this line and change all of the other `_session_expires` uses to look at the cookie in the session data itself instead.  Which certainly wont be as efficient but I think is necessary for interop.

atuttle replied:
I'll look into it. My initial impression was that express-session sets the TTL on its Redis inserts so that Redis removes them when they're expired and it doesn't have to do anything to expire them. Maybe Sidecar could/should do the same?

ryanguill replied:
We dont for two reasons - one because we store the session data as separate keys in the hash we want the session to expire all together, not some keys but not others.  But also because by keeping track of our own expiration it allows us to give the `sessionEndCallback` handler.

atuttle committed:
Merge branch 'master' into express-session-compat

atuttle committed:
Merge branch 'master' into express-session-compat

atuttle committed:
Remove _session_expires & rely on Redis TTL
Also documents trade-offs made for compatibility
Does include a backwards-compat change to sidecar.cfc

atuttle committed:
not all inputs will be strings

ryanguill committed:
small formatting changes to match style of existing code.
Removed unused functions in express_redis_session_store
Added a few extra notes in the README re: tradeoffs and alternatives.

ryanguill commented:
this LGTM @atuttle - if you want to review my changes just to make sure you don't have any objections, but once you give me the thumbs up ill go ahead an merge.

Thanks for the contribution and glad to have you using the project!
